app/api/resources/mentorship_relation.py

```diff
@@ -122,7 +122,7 @@ def put(cls, request_id):
 
         return response
 
-
+# TODO read cancellation reason from request body
```

Comment:
Can you help me with this @isabelcosta ?
`put('/mentorship_relation/%s/cancel' % self.mentorship_relation.id, headers=get_test_request_header(self.first_user.id))`
How do I send the `cancellation_reason` here in a PUT request?

Reply:
@bhawesh96 here's what I think you can do:

You have to create a model in this file `app/api/models/mentorship_relation.py` just with the `cancellation_reason` being optional. This will be used to tell the API that `cancellation_reason` is expected. Then don't forget to add it to the function `add_models_to_namespace` in that same file, otherwise that won't be on Swagger and can fail when testing the api.

To make the API wait for that model you will create you'll modify [this line](https://github.com/systers/mentorship-backend/blob/develop/app/api/resources/mentorship_relation.py#L132) that is above the `put` function:
```
@mentorship_relation_ns.expect(auth_header_parser, <model_you_just_created>)
``` 

Take a look [here](https://github.com/systers/mentorship-backend/blob/develop/app/api/models/user.py#L175) on how to make `cancellation_reason` optional. Tip: `required=False`

Then follow this example function `create_mentorship_relation` from `app/api/dao/mentorship_relation.py` file ([here](https://github.com/systers/mentorship-backend/blob/develop/app/api/dao/mentorship_relation.py#L14))
You want to receive json data inside the `cancel_relation` DAO funtion and then, if a `cancellation_reason` is sent you want to set it like [this](https://github.com/systers/mentorship-backend/blob/develop/app/api/dao/mentorship_relation.py#L217)
```
request.cancellation_reason = <cancellation_reason that came from PUT API>
```